pkg/visor/config.go

```diff
@@ -1,63 +1,49 @@
 package visor
 
 import (
-	"encoding/json"
 	"errors"
 	"fmt"
+	"net"
 	"os"
 	"path/filepath"
 	"time"
 
 	"github.com/SkycoinProject/dmsg"
 	"github.com/SkycoinProject/dmsg/cipher"
-	"github.com/SkycoinProject/dmsg/disc"
 	"github.com/SkycoinProject/dmsg/dmsgpty"
 
+	"github.com/SkycoinProject/skywire-mainnet/internal/skyenv"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/app/appcommon"
 	"github.com/SkycoinProject/skywire-mainnet/pkg/routing"
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet"
 	"github.com/SkycoinProject/skywire-mainnet/pkg/transport"
 	trClient "github.com/SkycoinProject/skywire-mainnet/pkg/transport-discovery/client"
 )
 
+const (
+	// DefaultTimeout is used for default config generation and if it is not set in config.
+	DefaultTimeout = Duration(10 * time.Second)
+	// DefaultLocalPath is used for default config generation and if it is not set in config.
+	DefaultLocalPath = "./local"
+	// DefaultAppsPath is used for default config generation and if it is not set in config.
+	DefaultAppsPath = "./apps"
+	// DefaultLogLevel is used for default config generation and if it is not set in config.
+	DefaultLogLevel = "info"
+	// DefaultSTCPPort ???
+	// TODO: Define above or remove below.
+	DefaultSTCPPort = 7777
+)
+
 // Config defines configuration parameters for Visor.
-// TODO(evanlinjin): Instead of having nested structs, make separate types for each field.
-// TODO(evanlinjin): Use pointers to allow nil-configs for non-crucial fields.
 type Config struct {
-	Version string `json:"version"`
-
-	Visor struct {
-		StaticPubKey cipher.PubKey `json:"static_public_key"`
-		StaticSecKey cipher.SecKey `json:"static_secret_key"`
-	} `json:"visor"`
-
-	STCP struct {
-		PubKeyTable map[cipher.PubKey]string `json:"pk_table"`
-		LocalAddr   string                   `json:"local_address"`
-	} `json:"stcp"`
-
-	Dmsg struct {
-		Discovery     string `json:"discovery"`
-		SessionsCount int    `json:"sessions_count"`
-	} `json:"dmsg"`
-
-	DmsgPty *DmsgPtyConfig `json:"dmsg_pty,omitempty"`
-
-	Transport struct {
-		Discovery string `json:"discovery"`
-		LogStore  struct {
-			Type     string `json:"type"`
-			Location string `json:"location"`
-		} `json:"log_store"`
-	} `json:"transport"`
-
-	Routing struct {
-		SetupNodes         []cipher.PubKey `json:"setup_nodes"`
-		RouteFinder        string          `json:"route_finder"`
-		RouteFinderTimeout Duration        `json:"route_finder_timeout,omitempty"`
-	} `json:"routing"`
-
-	Uptime struct {
-		Tracker string `json:"tracker"`
-	} `json:"uptime"`
+	Version       string               `json:"version"`
+	Visor         *KeyPair             `json:"visor"`
```

Comment:
Can we rename it to something like `KeyPair` or `Keys`? I mean the struct field itself. Cause it's like visor config having visor field. Which is just a pair of keys. Looks confusing

Reply:
@Darkren That's a good idea, had this in mind and forgot to do. However, I'm going to leave it in JSON as `visor` for now to not break configs. There're other names in the config that should also be changed. I'm going to come back to this later.